core/src/main/java/io/cloudevents/SpecVersion.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.cloudevents;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum SpecVersion {
+    V03(
+        "0.3",
+        Collections.unmodifiableSet(new HashSet<>(Arrays.asList("specversion", "id", "type", "source"))),
+        Collections.unmodifiableSet(new HashSet<>(Arrays.asList("datacontenttype", "datacontentencoding", "schemaurl", "subject", "time")))
+    ),
+    V1(
+        "1.0",
+        Collections.unmodifiableSet(new HashSet<>(Arrays.asList("specversion", "id", "type", "source"))),
+        Collections.unmodifiableSet(new HashSet<>(Arrays.asList("datacontenttype", "dataschema", "subject", "time")))
+    );
+
+    private final String stringValue;
+    private final Set<String> mandatoryAttributes;
+    private final Set<String> optionalAttributes;
+    private final Set<String> allAttributes;
+
+    SpecVersion(String stringValue, Set<String> mandatoryAttributes, Set<String> optionalAttributes) {
+        this.stringValue = stringValue;
+        this.mandatoryAttributes = mandatoryAttributes;
+        this.optionalAttributes = optionalAttributes;
+        this.allAttributes = Collections.unmodifiableSet(
+            Stream.concat(mandatoryAttributes.stream(), optionalAttributes.stream()).collect(Collectors.toSet())
+        );
+    }
+
+    @Override
+    public String toString() {
+        return this.stringValue;
+    }
+
+    public static SpecVersion parse(String sv) {
+        switch (sv) {
+            case "0.3":
+                return SpecVersion.V03;
+            case "1.0":
+                return SpecVersion.V1;
+            default:
+                throw new IllegalArgumentException("Unrecognized SpecVersion " + sv);
```

Comment:
Just an idea:
What if instead of throwing we return `SpecVersion.UNKNOWN` (does not allow any attribute)?
This way, it will be a little bit easier to deal with this method from functional code, plus there will be no need to know which exact exception type is thrown.

If not, we should at leat add `@throws` to method's javadoc (or maybe introduce a custom checked type then)

Reply:
I don't think it's a good idea, I'm not prone adding a new variant to the enum for "UNKNOWN", only because we don't want to fail the parsing process with an exception. I could rather return a null if it's not found, but adding an enum variant it's not correct imo.

About the javadoc, i didn't touched any of them https://github.com/cloudevents/sdk-java/pull/155#discussion_r429084873

Reply:
Ok i rewrote the javadocs for this method